easygui/boxes/egstore.py

```diff
@@ -6,73 +6,64 @@
-::
+        class Settings(EgStore):
+            def __init__(self, filename):  # filename is required
+                # specify default values for variables that this application wants to remember
+                self.user_id = ''
+                self.target_server = ''
 
-    class Settings(EgStore):
-        def __init__(self, filename):  # filename is required
-            #-------------------------------------------------
-            # Specify default/initial values for variables that
-            # this particular application wants to remember.
-            #-------------------------------------------------
-            self.userId = ""
-            self.targetServer = ""
+                # call super **after** setting defaults
+                super(Settings, self).__init__(filename)
 
-            #-------------------------------------------------
-            # For subclasses of EgStore, these must be
-            # the last two statements in  __init__
-            #-------------------------------------------------
-            self.filename = filename  # this is required
-            self.restore()            # restore values from the storage file if possible
+    **Example B: create settings, a persistent Settings object** ::
 
-**Example B: create settings, a persistent Settings object**
-::
+        settings = Settings('app_settings')
+        settings.user_id = 'obama_barak'
+        settings.targetServer = 'whitehouse1'
+        settings.store()
 
-    settingsFile = "myApp_settings.txt"
-    settings = Settings(settingsFile)
+        # run code that gets a new value for user_id, and persist the settings
+        settings.user_id = 'biden_joe'
+        settings.store()
 
-    user    = "obama_barak"
-    server  = "whitehouse1"
-    settings.userId = user
-    settings.targetServer = server
-    settings.store()    # persist the settings
+    **Example C: recover the Settings instance, change an attribute, and store it again.** ::
 
-    # run code that gets a new value for userId, and persist the settings
-    user    = "biden_joe"
-    settings.userId = user
-    settings.store()
+        settings = Settings('app_settings')
+        settings.user_id = 'vanrossum_g'
+        settings.store()
+    """
 
-**Example C: recover the Settings instance, change an attribute, and store it again.**
-::
+    def __init__(self, filename):
+        """Initialize a store with the given filename and try to load stored values.
+        If the filename doesn't exist yet, the restore is skipped.
 
-    settings = Settings(settingsFile)
-    settings.userId = "vanrossum_g"
-    settings.store()
+        Only attributes defined here will be stored.
+        Subclasses should initialize any attributes they want to store here, then call ``super``.
 
-"""
+        :param filename: the file that backs this store for saving and loading
+        """
+
+        self._filename = filename
```

Comment:
Use `_filename` rather than `_filename`, since the attribute is internal.  If you think this is a backwards compatibility issues, I can add a property that shows a deprecation warning.